ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1482,8 +1522,11 @@ protected void maybeProcessCallbacks() {
               if (!resolveRange(totalSize)) {
                 routerMetrics.getFirstDataChunkLatencyMs.update(System.currentTimeMillis() - submissionTimeMs);
                 routerMetrics.getDataChunkLatencyMs.update(System.currentTimeMillis() - initializedTimeMs);
-                chunkIndexToBuf.put(0, filterChunkToRange(decryptedBlobContent));
-                numChunksRetrieved.set(1);
+                ByteBuf decompressedContent = decompressContent(decryptedBlobContent);
+                if (decompressedContent != null) {
+                  chunkIndexToBuf.put(0, filterChunkToRange(decompressedContent));
+                  numChunksRetrieved.set(1);
+                }
```

Comment:
If decompression fails and `decompressedContent != null` is false, then what should we do?

Reply:
If decompression failed, we set the operation status to failed with the exception.  Setting the operation status and exception is done inside the decompressContent().